### PR TITLE
Add an option to the uncompressed zlib encoder.

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: lib/zlib.toit [hunk 1]
```diff
@@ -33,25 +33,35 @@ class UncompressedDeflateBackEnd_ implements BackEnd_:
   buffer_/ByteArray? := ?        // When this is null, everything has been read.
   buffer-fullness_/int := ?
   position-of-block-header_/int := ?
+  split-writes/bool
 
-  constructor .summer_ --gzip-header/bool:
+  constructor .summer_ --gzip-header/bool --.split-writes=true:
```
Comment (kasperl) [inline lib/zlib.toit:38]: Not sure I would have a default value here.

File: lib/zlib.toit [hunk 2]
```diff
@@ -91,9 +102,15 @@ Creates an uncompressed data stream that is compatible with zlib decoders
...
+  /**
+  Normally the literal blocks in the output will bear no releation
+    to the write operations on this encoder.  This reduces peak
+    memory use.  However, if $split-writes is true then a single
+    write operation will never span two blocks.
+  */
+  constructor --split-writes/bool=true:
     super
-        UncompressedDeflateBackEnd_ adler32.Adler32 --gzip-header=false
+        UncompressedDeflateBackEnd_ adler32.Adler32 --gzip-header=false --split-writes=split-writes
```
Comment (kasperl) [inline lib/zlib.toit:113]: BackEnd looks pretty weird to me. I think backend is way more common.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/zlib.toit [hunk 3]
```diff
@@ -91,9 +102,15 @@ Creates an uncompressed data stream that is compatible with zlib decoders
   be used from different tasks to prevent deadlocks.
 */
 class UncompressedZlibEncoder extends Coder_:
-  constructor:
+  /**
+  Normally the literal blocks in the output will bear no releation
```
Comment (floitsch) [inline lib/zlib.toit:106]:
```suggestion
  Normally the literal blocks in the output will bear no relation
```

**Review by floitsch — APPROVED** [review]
LGTM.

File: lib/zlib.toit [hunk 4]
```diff
@@ -28,30 +28,40 @@ MINIMAL-GZIP-HEADER_ ::= [0x1f, 0x8b, 8, 0, 0, 0, 0, 0, 0, 0xff]
...
-class UncompressedDeflateBackEnd_ implements BackEnd_:
+class UncompressedDeflateBackend_ implements Backend_:
   summer_/crypto.Checksum? := ?  // When this is null, we are closed for write.
   buffer_/ByteArray? := ?        // When this is null, everything has been read.
   buffer-fullness_/int := ?
   position-of-block-header_/int := ?
+  split-writes/bool
 
-  constructor .summer_ --gzip-header/bool:
+  constructor .summer_ --gzip-header/bool --.split-writes:
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply lib/zlib.toit:38]: It's documented in the user-facing public class.

**Pull request merged**